### ISSUE TITLE
Settings Page item selection bug using Xbox controller in Navigation Pane Template

### ISSUE THREAD

**Issue by shaheedmalik** [microsoft/WindowsTemplateStudio#3407]
**Describe the bug**
There is a item selection error in the Settings Page in the Navigation Pane C# Universal Windows Navigation Pane on the Settings page

**To Reproduce**
1.  Create a new Navigation Pane template with Code Behind and a Setting page.'.'
2. Using a Xbox controller navigate to the settings page'....'
3. Trying scrolling up and down using the controller.
4. See how the selected item jumps to the menu?

![App72019-10-3011-15-37_Trim](https://user-images.githubusercontent.com/51347950/67874021-b798cd00-fb0a-11e9-8620-6a4c1ccfdeaa.gif)

**Expected behavior**
The selected item should go from Light to Dark to System

![App72019-10-3011-15-37_Trim2](https://user-images.githubusercontent.com/51347950/67873795-6ab4f680-fb0a-11e9-99ef-a41c5a13a485.gif)

**Screenshots**

**Additional context**

**System**

- **VS Version:** C++ 2019   00435-60000-00000-AA805
- **WinTS Wizard Version:** 1.0
- **WinTS Template Version:** 3.4.19254.1
- **Windows Build:** 1862.10024

**Comment by sibille** [comment]
Thanks for reporting! We'll have a look at this, seems the theme is not getting properly selected

**Comment by shaheedmalik** [comment]
> Thanks for reporting! We'll have a look at this, seems the theme is not getting properly selected

Oddly enough, it works perfectly fine with the keyboard.

**Comment by mrlacey** [comment]
I have a quick, hacky workaround (which may or may not be a solution) but will investigate if this impacts other pages and if it's something lower level that's got a problem that we're seeing.
I also need to investigate if there are any possible side-effects to my solution.

**Comment by mrlacey** [comment]
This affects more than just the settings page

- mediplayer - can't nav down to controls
- webview - can't navigate within page or get out of page or to nav at bottom
- TreeView can't get to page
- DataGrid can't navigate outside of grid (back to nav)
- tabview - can't delete tabs
- map - displays dialog that can't interact with with controller
- camera - unusable dialog
- ink pages (all 3) - can't get out of buttons at top

**Comment by mrlacey** [comment]
@shaheedmalik I'll add a fix for the issue you reported and some of the others too.
Some of the issues I'm finding on other pages seem to be down to problems with the individual controls.
To help me escalate the importance of the issue, could you please explain what you're using the controller for in a Windows 10 app? (Having real use cases can help when prioritizing work--including fixes.)

**Comment by mrlacey** [comment]
@shaheedmalik you can work around the issue by adding a `ScrollViewer` (with `VerticalScrollBarVisibility` and `VerticalScrollMode` set to auto) around the `StackPanel` inside the main `Grid` on the page.
If that doesn't make sense, see [this example](https://github.com/mrlacey/WindowsTemplateStudio/commit/10eb31354b111dec141e6d4d49c4bf8a44fb0ed1#diff-45029f838d46ccc80683a4a0ed82c2ca).

**Comment by mrlacey** [comment]
After further discussion, we have decided not to make any changes to the generated code to account for this scenario.
There are multiple reasons for this.
- There is a workaround.
- It's a very niche scenario.
- It (the specific scenario covered by the OP) looks to be related to an underlying issue with the NavigationView control and a bug has been raised in the WinUI repo to look at that.
- Adding to the generated code would add complexity to everyone for scenarios that are only applicable to a few.
- Addressing this for all scenarios is a lot of work and I can't justify committing to it for the small number of people impacted.
- We don't have the ability to create and maintain automated tests to ensure that all future output will continue to work in this scenario.


The possible bug(s) linked above as exceptions, situations like this should be very hard to get into but there are ways that these can be addressed by end-developers on an as-needed basis. For more on this see the documentation at https://docs.microsoft.com/en-us/windows/uwp/design/input/gamepad-and-remote-interactions#xy-focus-navigation-and-interaction


I have moved this to the backlog but I'm planning on closing this in a few days unless there is a good argument to keep it open.
